### DISCUSSION TITLE
questions on maneuver , tripEdge and GetCurrEdge()

### DISCUSSION THREAD

**Discussion by mkandulavm** [valhalla/valhalla#4761] [Q&A]
Hi 

When iterating through maneuvers during ManeuversBuilder over its its shape indices, I am seeing that the GetCurrEdge() returns nullptr sometimes.

Why does this happen ? Is it for shortcuts ? Is it possible to get the base edges ? 

I am trying to get a list of way IDs that form the route, and the above is causing an issue.

Thank you.

**Comment by nilsnolde** [comment]
Let’s keep things like these in discussions, 600+ issues is enough;)

From memory I can tell you that shortcuts never make it that far, we recover those to base edges before they go to triplegbuilder.

**Reply by mkandulavm** [reply]
ok. So why would trip_path_->getCurrEdge() return null for a shape index ?
